ssg/oval_object_model/oval_document.py

```diff
@@ -62,6 +62,14 @@ def load_oval_document(oval_document_xml_el):
     return oval_document
 
 
+def _get_filtered_list_of_oval_components(component_dict, id_selection):
+    return [
+        component
+        for component in component_dict.values()
+        if component.id_ in id_selection
+    ]
```

Comment:
This seems to me to be an inefficient way of doing it. IIUC the keys of the `component_dict` directory are IDs of the components so these keys are equal to the value of the `id_` member of this dictionary's values. Therefore it doesn't make sense to iterate over all components and check their IDs. It would be better to iterate over IDs in the `id_selection` in first place and use each item of this iteration as key to `component_dict`. WDYT? Would that work?

Reply:
Yes, I changed the approach and used the generator to iterate and filter the OVAL components.